packages/block-library/src/navigation/index.php

```diff
@@ -67,6 +67,521 @@ function block_core_navigation_sort_menu_items_by_parent_id( $menu_items ) {
+			$script_handles = $block->block_type->view_script_handles;
+
+			// If the script is not needed, and it is still in the `view_script_handles`, remove it.
+			if ( ! $should_load_view_script && in_array( $view_js_file, $script_handles, true ) ) {
+				$block->block_type->view_script_handles = array_diff( $script_handles, array( $view_js_file ) );
+			}
+			// If the script is needed, but it was previously removed, add it again.
+			if ( $should_load_view_script && ! in_array( $view_js_file, $script_handles, true ) ) {
+				$block->block_type->view_script_handles = array_merge( $script_handles, array( $view_js_file ) );
+			}
+		}
+	}
+
+	/**
+	 * Returns the markup for the navigation block.
+	 */
+	private static function get_nav_markup( $attributes, $inner_blocks ) {
+		$inner_blocks_html = WP_Navigation_Block::get_inner_blocks_html( $inner_blocks, $attributes );
+		if ( WP_Navigation_Block::is_responsive_navigation( $attributes ) ) {
+			return WP_Navigation_Block::get_responsive_container_markup( $attributes, $inner_blocks, $inner_blocks_html );
+		}
+		return $inner_blocks_html;
+	}
+
+	/**
+	 * Renders the navigation block.
+	 */
+	static function render( $attributes, $content, $block ) {
+		static $seen_menu_names = array();
+
+		$nav_menu_name = WP_Navigation_Block::get_navigation_name( $attributes, $seen_menu_names );
+
+		/**
+		 * Deprecated:
+		 * The rgbTextColor and rgbBackgroundColor attributes
+		 * have been deprecated in favor of
+		 * customTextColor and customBackgroundColor ones.
+		 * Move the values from old attrs to the new ones.
+		 */
+		if ( isset( $attributes['rgbTextColor'] ) && empty( $attributes['textColor'] ) ) {
+			$attributes['customTextColor'] = $attributes['rgbTextColor'];
+		}
+
+		if ( isset( $attributes['rgbBackgroundColor'] ) && empty( $attributes['backgroundColor'] ) ) {
+			$attributes['customBackgroundColor'] = $attributes['rgbBackgroundColor'];
+		}
+
+		unset( $attributes['rgbTextColor'], $attributes['rgbBackgroundColor'] );
+
+		$inner_blocks       = WP_Navigation_Block::get_inner_blocks_for_navigation( $block, $attributes );
+		// Prevent navigation blocks referencing themselves from rendering.
+		if ( block_core_navigation_block_contains_core_navigation( $inner_blocks ) ) {
+			return '';
+		}
+
+		// If the menu name has been used previously then append an ID
+		// to the name to ensure uniqueness across a given post.
+		if ( isset( $seen_menu_names[ $nav_menu_name ] ) && $seen_menu_names[ $nav_menu_name ] > 1 ) {
+			$count         = $seen_menu_names[ $nav_menu_name ];
+			$nav_menu_name = $nav_menu_name . ' ' . ( $count );
+		}
+
+		$wrapper_attributes = WP_Navigation_Block::get_nav_wrapper_attributes( $attributes, $inner_blocks, $nav_menu_name );
+
+		WP_Navigation_Block::handle_view_script_loading( $block, $attributes, $inner_blocks );
+
+		return sprintf(
+			'<nav %1$s>%2$s</nav>',
+			$wrapper_attributes,
+			WP_Navigation_Block::get_nav_markup( $attributes, $inner_blocks )
```

Comment:
```suggestion
			WP_Navigation_Block::get_wrapper_markup( $attributes, $inner_blocks )
```

Maybe let's use the established convention?

Reply:
done in 5157393ba7